specs/refactoring_of_node_monitoring_flows_into_node_agent.adoc

```diff
@@ -0,0 +1,224 @@
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/flows/configure_collectd/__init__.py
+----
+
+and a genric command to generate config file from template as in:
+
+----
+https://github.com/Tendrl/node-monitoring/blob/master/tendrl/node_monitoring/commands/config_manager.py
+----
+
+will be moved to node-agent.
+
+* The templates currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/templates
+----
+
+and the plugins currently in:
+
+----
+https://github.com/Tendrl/node-monitoring/tree/master/tendrl/node_monitoring/plugins
+----
+
+will be moved into node-agent.
+
+* Earlier the jobs for intiating the flows to configure collectd on nodes were
+  initiated by performance-monitoring node. But for now(momentarily) as a final
+  step of import, create and expand cluster, this will be intiated on all nodes
+  of cluster.
+
+* Initiating(loading job to etcd) configuration on cluster nodes will involve
+  invoking configuration flow from an atom marked post atom in import cluster
+  flow.
+  ----
+  Job loading sample:
+  https://github.com/Tendrl/performance-monitoring/blob/master/tendrl/performance_monitoring/utils/util.py#L41
+  ----
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+A new flow definition will be added to node-agent under its namespace as below:
+
+----
+namespace.node_agent:
+  flows:
+    ConfigureCollectd:
+      atoms:
+        - tendrl.objects.Node.atoms.Cmd
+      help: "Execute given command on given node"
+      enabled: true
+      inputs:
+        mandatory:
+          - Node.fqdn
+          - plugin_name
+          - plugin_conf_params
+          - Service.name
+      post_run:
+        - tendrl.objects.Service.atoms.CheckServiceStatus
+      pre_run:
+        - tendrl.objects.Node.atoms.CheckNodeUp
+      run: tendrl.node_agent.flows.ConfigureCollectd
+      type: Create
+      uuid: dc8fff3a-34d9-4786-9282-55eff6abb6c4
+      version: 1
+tendrl_schema_version: 0.3
+----
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+None
+
+==== Notifications/Monitoring impact:
+
+node-monitoring project will be deprecated.
+
+==== Tendrl/common impact:
+None
+
+==== Tendrl/node_agent impact:
+
+As described in section "Proposed Change" section, collectd templates, custom
+plugins and flow to generate collectd configuration will move into node-agent.
+
+* The plugins will go as under :
+
+----
+[node_agent]$ tree tendrl/node_agent/collectd/collectors/
+tendrl/node_agent/collectd/collectors/
+├── glusterfs
+│   ├── __init__.py
+│   ├── tendrl_glusterfs_brick_utilization.py
+│   ├── tendrl_glusterfs_health_counters.py
+│   ├── tendrl_glusterfs_peer_network_throughput.py
+│   ├── tendrl_glusterfs_profile_info.py
+│   └── utils.py
+├── ini2json.py
+├── __init__.py
+├── tendrl_ceph_cluster_iops.py
+├── tendrl_ceph_node_network_throughput.py
+├── tendrl_ceph_utilization.py
+├── tendrl_glusterfs_base.py
+└── tendrl_glusterfs_clients_info.py
```

Comment:
Ack. The only reason was that, the plugins are classified/grouped and a base class loads all plugins within a specific directory at once at fixed interval of time.But then, the problem here is, some plugins cannot be executed as frequently as a few others..

So, as the classification of plugins possibly into heavy-weight(less frequently executed) and light -weight(more frequently executed) is currently not implemented, I did it this way.. I'll change it nowAck. The only reason was that, the plugins are classified/grouped and a base class loads all plugins within a specific directory at once at fixed interval of time.But then, the problem here is, some plugins cannot be executed as frequently as a few others..

So, as the classification of plugins possibly into heavy-weight(less frequently executed) and light -weight(more frequently executed) is currently not implemented, I did it this way.. I'll change it now